llama_index/llms/openai.py

```diff
@@ -313,18 +316,21 @@ def gen() -> CompletionResponseGen:
 
         return gen()
 
-    def _update_max_tokens(self, all_kwargs: Dict[str, Any], prompt: str) -> None:
-        if self.max_tokens is not None:
-            return
-        # NOTE: non-chat completion endpoint requires max_tokens to be set
+    def _get_encoding_for_model(self):
```

Comment:
Yeah check my other comment on using the `typing.Protocol` with `_Tokenizer` as the return type.

Reply:
You're right here though, it can't be used as a type hint.  So we use a `typing.Protocol` to allow for anything supporting that signature

Reply:
isn't tiktoken a dependency of llama-index though? So it should be installed?

Reply:
Oh you're right, it can be used, I should have checked the core LlamaIndex dependencies.  Even so, for subclasses not using `tiktoken` (e.g. `OpenAILike`), it should just be a `typing.Protocol` for any object supporting an `encode` function